src/main/java/org/ibp/api/java/impl/middleware/inventory_new/LotServiceImpl.java

```diff
@@ -14,16 +20,45 @@
 @Transactional
 public class LotServiceImpl implements LotService {
 
+	@Autowired
+	private LotInputValidator lotInputValidator;
+
+	@Autowired
+	private SecurityService securityService;
+
 	@Autowired
 	private org.generationcp.middleware.service.api.inventory.LotService lotService;
 
+	@Autowired
+	private StockService stockService;
+
+	private static final String DEFAULT_STOCKID_PREFIX = "SID";
+
 	@Override
-	public List<LotDto> searchLots(final LotsSearchDto lotsSearchDto, final Pageable pageable) {
+	public List<ExtendedLotDto> searchLots(final LotsSearchDto lotsSearchDto, final Pageable pageable) {
 		return lotService.searchLots(lotsSearchDto, pageable);
 	}
 
 	@Override
 	public long countSearchLots(final LotsSearchDto lotsSearchDto) {
 		return lotService.countSearchLots(lotsSearchDto);
 	}
+
+	@Override
+	public Integer saveLot(final LotGeneratorInputDto lotGeneratorInputDto) {
+		final WorkbenchUser loggedInUser = this.securityService.getCurrentlyLoggedInUser();
+		lotGeneratorInputDto.setUserId(loggedInUser.getUserid());
+		lotInputValidator.validate(lotGeneratorInputDto);
+		final String nextStockIDPrefix;
```

Comment:
@vmaletta does this var need to be declared before the if? I think it is not used outside, right?

Reply:
@clarysabel you're right